Source/BugsnagEvent.m

```diff
@@ -385,22 +398,22 @@ - (instancetype)initWithKSReport:(NSDictionary *)report {
     return self;
 }
 
-- (instancetype _Nonnull)
-initWithErrorName:(NSString *_Nonnull)name
-     errorMessage:(NSString *_Nonnull)message
-    configuration:(BugsnagConfiguration *_Nonnull)config
-         metadata:(NSDictionary *_Nonnull)metadata
-     handledState:(BugsnagHandledState *_Nonnull)handledState
-          session:(BugsnagSession *_Nullable)session {
+- (instancetype _Nonnull)initWithErrorName:(NSString *_Nonnull)name
+                              errorMessage:(NSString *_Nonnull)message
+                             configuration:(BugsnagConfiguration *_Nonnull)config
+                                  metadata:(BugsnagMetadata *_Nullable)metadata
+                              handledState:(BugsnagHandledState *_Nonnull)handledState
+                                   session:(BugsnagSession *_Nullable)session
+{
     if (self = [super init]) {
         _errorClass = name;
         _errorMessage = message;
         _overrides = [NSDictionary new];
-        _metadata = metadata ?: [NSDictionary new];
+        self.metadata = metadata ?: [BugsnagMetadata new];
```

Comment:
There's a bit of a gotcha here in that the `metadata` passed in by the `BugsnagClient` will continue to be mutated. I'd feel better about making a defensive copy in the constructor than at each call site, as it reduces the risk that someone forgets to add in a call to `deepCopy` sometime in the future.

Reply:
Agreed, although at present there was only one substantive call, you're right putting the defensive copy in a single place is preferred.